src/main/java/com/minecolonies/coremod/client/gui/WindowMultiBlock.java

```diff
@@ -0,0 +1,295 @@
+        this.init();
+    }
+
+    private void init()
+    {
+        //Register all necessary buttons with the window.
+        registerButton(BUTTON_CONFIRM, this::confirmClicked);
+        registerButton(BUTTON_CANCEL, this::cancelClicked);
+        registerButton(BUTTON_LEFT, this::moveLeftClicked);
+        registerButton(BUTTON_RIGHT, this::moveRightClicked);
+        registerButton(BUTTON_UP, this::moveUpClicked);
+        registerButton(BUTTON_DOWN, this::moveDownClicked);
+        registerButton(BUTTON_BACKWARD, this::moveBackClicked);
+        registerButton(BUTTON_FORWARD, this::moveForwardClicked);
+    }
+
+    /**
+     * Called when the window is opened.
+     * Sets up the buttons for either hut mode or decoration mode.
+     */
+    @Override
+    public void onOpened()
+    {
+        final TileEntity block = Minecraft.getMinecraft().world.getTileEntity(pos);
+        if (block instanceof TileEntityMultiBlock)
+        {
+            inputRange.setText(Integer.toString(((TileEntityMultiBlock) block).getRange()));
+            inputSpeed.setText(Integer.toString(((TileEntityMultiBlock) block).getSpeed()));
+            final EnumFacing dir = ((TileEntityMultiBlock) block).getDirection();
+            final EnumFacing out = ((TileEntityMultiBlock) block).getOutput();
+            enable(dir, dir, false);
+            enable(out, out, true);
+            return;
+        }
+        close();
+    }
+
+    private void enable(final EnumFacing oldFacing, final EnumFacing newFacing, final boolean rightClick)
+    {
+        switch (oldFacing)
+        {
+            case UP:
+                findPaneOfTypeByID(BUTTON_UP, ButtonImage.class).setImage(new ResourceLocation(Constants.MOD_ID, String.format(RES_STRING, BUTTON_UP)));
+                break;
+            case DOWN:
+                findPaneOfTypeByID(BUTTON_DOWN, ButtonImage.class).setImage(new ResourceLocation(Constants.MOD_ID, String.format(RES_STRING, BUTTON_DOWN)));
+                break;
+            case NORTH:
+                findPaneOfTypeByID(BUTTON_FORWARD, ButtonImage.class).setImage(new ResourceLocation(Constants.MOD_ID, String.format(RES_STRING, BUTTON_FORWARD)));
+                break;
+            case SOUTH:
+                findPaneOfTypeByID(BUTTON_BACKWARD, ButtonImage.class).setImage(new ResourceLocation(Constants.MOD_ID, String.format(RES_STRING, BUTTON_BACKWARD)));
+                break;
+            case EAST:
+                findPaneOfTypeByID(BUTTON_RIGHT, ButtonImage.class).setImage(new ResourceLocation(Constants.MOD_ID, String.format(RES_STRING, BUTTON_RIGHT)));
+                break;
+            case WEST:
+                findPaneOfTypeByID(BUTTON_LEFT, ButtonImage.class).setImage(new ResourceLocation(Constants.MOD_ID, String.format(RES_STRING, BUTTON_LEFT)));
+                break;
+        }
+
+        final String color = rightClick ? RED_POS : GREEN_POS;
+        switch (newFacing)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)